Disable 'dist' and 'source-dist' make targets

nalimilan opened the pull request:
Yet another step towards making 'dist' equivalent to 'light-source-dist', as in most projects.

Cf. https://github.com/JuliaLang/julia/pull/10253. I could have first added the warning and still called the real target, but I don't think many people would notice it, would they? The fix is quite easy to apply.

Next step will be to change the meaning of `dist`, when everybody will have switched (probably before 0.4?).

CC: @tkelman @staticfloat

nalimilan commented:
(Cancelled AppVeyor build manually.)

staticfloat commented:
Is the purpose of this just to disable `dist`?  I think it makes more sense to just move over to having `dist` do the new behavior immediately.  I mean, between you, me and @tkelman I think we've got a good fraction of the people in the world that would care about its meaning changing.

tkelman commented:
I'm mainly concerned about the buildbots (good fraction of the people again), and hitting the buttons on them to build release-0.3 as well as automatic nightlies. I think I backported the new meaning of `binary-dist` to release-0.3 already, right?

nalimilan commented:
@tkelman Yes, you backported it in https://github.com/JuliaLang/julia/commit/0d1e7bcf261e4bad1587d55c929b7647b41215b9 So the buildbots can be updated any time before merging this.

@staticfloat I don't really care, but there have been bug reports from people using `make dist` to build Julia an use it later on machines without Internet access (or whatever). Since there's no hurry to get the new meaning of `make dist`, I figured it wouldn't hurt to show the error for some time to limit confusion.

staticfloat commented:
Yes, I [updated the buildbots](https://github.com/staticfloat/julia-buildbot/commit/d1c87c23f2b9d76d53caf5d98ba11181de8dd254) last night to ensure they would continue to work with these changes.

Let's make sure to add documentation changes to this PR.  Specifically, `DISTRIBUTING.md` will need some love.

nalimilan committed:
Disable 'dist' and 'source-dist' make targets

Yet another step towards making 'dist' equivalent to
'light-source-dist', as in most projects. Also
change mentions of old targets in docs.

[av skip]

nalimilan commented:
I've updated the PR, fixing all occurrences of `make dist` and of `source-dist` I could find.

tkelman commented:
Looks like we're good here.

tkelman commented:
missed one https://github.com/JuliaLang/julia/blob/cfbd68c81f8f7e49ec92f5775666ff5bb6b1fe76/contrib/mac/app/Makefile#L19
